mongodb/src/main/scala/com/stratio/crossdata/connector/mongodb/MongoQueryProcessor.scala

```diff
@@ -38,47 +42,66 @@ object MongoQueryProcessor {
-        queryBuilder.put(attribute).greaterThanEquals(convertToStandardType(value))
+        queryBuilder.put(attribute).greaterThanEquals(checkObjectID(attribute, value, config))
       case sources.In(attribute, values) =>
-        queryBuilder.put(attribute).in(values.map(convertToStandardType))
+        queryBuilder.put(attribute).in(values.map(value => checkObjectID(attribute, value, config)))
       case sources.LessThan(attribute, value) =>
-        queryBuilder.put(attribute).lessThan(convertToStandardType(value))
+        queryBuilder.put(attribute).lessThan(checkObjectID(attribute, value, config))
       case sources.LessThanOrEqual(attribute, value) =>
-        queryBuilder.put(attribute).lessThanEquals(convertToStandardType(value))
+        queryBuilder.put(attribute).lessThanEquals(checkObjectID(attribute, value, config))
       case sources.IsNull(attribute) =>
         queryBuilder.put(attribute).is(null)
       case sources.IsNotNull(attribute) =>
         queryBuilder.put(attribute).notEquals(null)
-      case sources.And(leftFilter, rightFilter) =>
-        queryBuilder.and(filtersToDBObject(Array(leftFilter)), filtersToDBObject(Array(rightFilter)))
-      case sources.Or(leftFilter, rightFilter) =>
-        queryBuilder.or(filtersToDBObject(Array(leftFilter)), filtersToDBObject(Array(rightFilter)))
-      case sources.StringStartsWith(attribute, value) =>
+      case sources.And(leftFilter, rightFilter) if !parentFilterIsNot =>
+        queryBuilder.and(filtersToDBObject(config, Array(leftFilter)), filtersToDBObject(config, Array(rightFilter)))
+      case sources.Or(leftFilter, rightFilter) if !parentFilterIsNot =>
+        queryBuilder.or(filtersToDBObject(config, Array(leftFilter)), filtersToDBObject(config, Array(rightFilter)))
+      case sources.StringStartsWith(attribute, value) if !parentFilterIsNot =>
         queryBuilder.put(attribute).regex(Pattern.compile("^" + value + ".*$"))
-      case sources.StringEndsWith(attribute, value) =>
+      case sources.StringEndsWith(attribute, value) if !parentFilterIsNot =>
         queryBuilder.put(attribute).regex(Pattern.compile("^.*" + value + "$"))
-      case sources.StringContains(attribute, value) =>
+      case sources.StringContains(attribute, value) if !parentFilterIsNot =>
         queryBuilder.put(attribute).regex(Pattern.compile(".*" + value + ".*"))
-      // TODO Not filter
+      case sources.Not(filter) =>
+        filtersToDBObject(config, Array(filter), true)
     }
+
     queryBuilder.get
   }
 
-  private def convertToStandardType(value: Any): Any = value
+    /**
+      * Check if the field is "_id" and if the user wants to filter by this field as an ObjectId
+      *
+      * @param attribute Name of the file
+      * @param value Value for the attribute
+      * @return The value in the correct data type
+      */
+     private def checkObjectID(attribute: String, value: Any, config: Config) : Any = {
```

Comment:
Same here, this function has, how Pablo says?, Collateral effects? the name indicates just a verification, but builds a new Object.